### PR TITLE
fix format string.

### PR DESCRIPTION
**Pull request by btuduri**
Hello,

This line seems make some warning with clang, so i fixed them.

Sincerly,

### CODE REVIEW

**Comment by fincs** [comment]
This is wrong on 32-bit systems.

**Comment by btuduri** [comment]
oh ok, but I will checking on a Gnu Linux Debian to be sure. This warning was reported with my osx clang and xcode 6.1.1.

File: src/gdb/gdbstub.c [hunk 1]
```diff
@@ -927,13 +927,13 @@ processPacket_gdb( SOCKET_TYPE sock, const uint8_t *packet,
       }
       if (strcmp((char*)packet, "qSupported") == 0) {
           /* query of supported features */
-          sprintf((char*)out_ptr, "PacketSize=%u;ReverseContinue-;ReverseStep-", sizeof(packet)); //it is how it is
+          sprintf((char*)out_ptr, "PacketSize=%lu;ReverseContinue-;ReverseStep-", sizeof(packet)); //it is how it is
```
Comment (mtheall) [inline src/gdb/gdbstub.c]: Use %zu for size_t

File: src/gdb/gdbstub.c [hunk 2]
```diff
@@ -927,13 +927,13 @@ processPacket_gdb( SOCKET_TYPE sock, const uint8_t *packet,
...
           send_size = strlen((char*)out_ptr);
           break;
       }
       if (strcmp((char*)packet, "qSupported:qRelocInsn+") == 0) {
           /* query of supported features */
-          sprintf((char*)out_ptr, "PacketSize=%u;RelocInsn-;ReverseContinue-;ReverseStep-", sizeof(packet)); //it is how it is
+          sprintf((char*)out_ptr, "PacketSize=%lu;RelocInsn-;ReverseContinue-;ReverseStep-", sizeof(packet)); //it is how it is
```
Comment (mtheall) [inline src/gdb/gdbstub.c]: Use %zu for size_t

File: src/services/fs_user.c [hunk 3]
```diff
@@ -930,7 +930,7 @@ SERVICE_CMD(0x08040000)   // GetSize
         rc = -1;
     }
 
-    DEBUG("GetSize ret=%08X(%d)\n",sz,sz);
+    DEBUG("GetSize ret=%08llX(%lld)\n",sz,sz);
```
Comment (mtheall) [inline src/services/fs_user.c]: Try this:
#include <inttypes.h>
DEBUG("GetSize ret=%08" PRIX64 "(%" PRId64 ")\n",sz,sz);

File: src/syscalls/timer.c [hunk 4]
```diff
@@ -56,7 +56,7 @@ u32 svcSetTimer()
     u32 timer = arm11_R(0);
     u64 initial = arm11_R(1) | ((u64)arm11_R(2) << 32);
     u64 interval = arm11_R(3) | ((u64)arm11_R(4) << 32);
-    DEBUG("SetTimer --todo-- %08x %08x %08x\n", timer, initial, interval);
+    DEBUG("SetTimer --todo-- %08x %08llx %08llx\n", timer, initial, interval);
```
Comment (mtheall) [inline src/syscalls/timer.c]: Try this:
    #include <inttypes.h>
    DEBUG("SetTimer --todo-- %08x %08" PRIx64 " %08 "PRIx64 "\n", timer, initial, interval);

File: src/gdb/gdbstub.c (hunk 2 above)
Comment (btuduri) [inline src/gdb/gdbstub.c]: I'm agree. :) I haven't use %zu because I dunno if it will supported by MinGW.

File: src/services/fs_user.c (hunk 3 above)
Comment (btuduri) [inline src/services/fs_user.c]: The compilers returns : 
src/services/fs_user.c:933:5: note: to match this '('
    DEBUG("GetSize ret=%08" PRIX64 "(%" PRId64 ")\n",sz,sz);
    ^

In the code I write : 

DEBUG("GetSize ret=%08" PRIX64 "(%" PRId64 ")\n",sz,sz);

File: src/syscalls/timer.c (hunk 4 above)
Comment (btuduri) [inline src/syscalls/timer.c]: src/syscalls/timer.c:59:40: error: expected ')'
    DEBUG("SetTimer --todo-- %08x %08" PRIx64 " %08" PRIx64 "\n", timer, initial, interval);
                                       ^
src/syscalls/timer.c:59:5: note: to match this '('
    DEBUG("SetTimer --todo-- %08x %08" PRIx64 " %08" PRIx64 "\n", timer, initial, interval);
    ^


I wrote :
DEBUG("SetTimer --todo-- %08x %08" PRIx64 " %08" PRIx64 "\n", timer, initial, interval);

**Comment by fincs** [comment]
`%zu` is not supported on Windows (VC++, not sure about MinGW) for whatever reason, instead it's called `%Iu`. It may be necessary to `#ifdef _MSC_VER` the appropriate printf flag.

**Comment by btuduri** [comment]
@fincs you confirm my doubt with %zu for MSVC, That's why I used the %lu while %zu is semantically correct.

**Comment by btuduri** [comment]
Else a good alternative will be to use clang with MSVC ?

**Comment by btuduri** [comment]
by the way, the -lpthread is required when i compile on my gnu debian . And for you not ?

I have the same effect for clang on linux.
https://gist.github.com/btuduri/f835fe034836d9e62705

**Comment by mtheall** [comment]
Don't use -lpthread, use -pthread. It should compile fine without it, but the pthread functions would be stubs that aren't truly multithreading.

File: src/syscalls/timer.c (hunk 4 above)
Comment (mtheall) [inline src/syscalls/timer.c]: You have to add #include &lt;inttypes.h&gt;

**Comment by btuduri** [comment]
it's understand @mtheall :)

**Pull request closed without merging**